### DISCUSSION TITLE
How can I post a file to an endpoint?

### DISCUSSION THREAD

**Discussion by crysfel** [sindresorhus/got#1894] [Q&A] [answered]
#### What would you like to discuss?

Hi there! I'm trying to post an mp3 file along with other data to a REST endpoint, but I can't find an example of how to do that. Is it even possible?

#### Checklist

- [x] I have read the documentation.

**Comment by sindresorhus** [comment, marked as answer]
It really depends on what the endpoint accepts. There's not enough information here to answer you specifically. The `body` options accepts a [`FormData` instance](https://github.com/sindresorhus/got/blob/a485234df6b741393310f7341c256fa43aa5d3a4/documentation/2-options.md#body), which can often be used for this purpose.
